src/controllers/keyboard/keyboardcontrollerpreset.cpp

```diff
@@ -0,0 +1,242 @@
+        // translation it could potentially contain multiple key sequences.
+        QStringList keyseqs = QStringList() += (!keyseqsRaw.isEmpty() ? keyseqsRawI->keysequence : "");
+
+        if (keyseqNeedsTranslate) {
+
+            // Can't happen. Otherwise keyseqNeedsTranslate would be false.
+            DEBUG_ASSERT(!keyseqsRawI->byPositionOf.isEmpty());
+
+            QString key = layoutUtils::getCharFromKeysequence(keyseqs[0]);
+            if (key.size() > 1) {
+                qWarning() << "Can't translate keyseq" << keyseqs[0]
+                           << "because key" << key << "is not a character.";
+            }
+
+            // Get original layout
+            KeyboardLayoutPointer originalKeyseqLayout = getLayout(keyseqsRawI->byPositionOf.toStdString());
+            if (originalKeyseqLayout == nullptr) {
+                qWarning() << "Can't translate keyseq" << keyseqs[0]
+                           << "because Mixxx can't retrieve it's scancode."
+                           << "'byPositionOf' layout" << keyseqsRawI->byPositionOf
+                           << "not found. Skipping mapping for" << configKey;
+                continue;
+            }
+
+            // Check whether we need no modifier or shift modifier for the scancode lookup
+            QStringList modifiers = layoutUtils::getModifiersFromKeysequence(keyseqs[0]);
+            bool onlyShift = modifiers.size() == 1 && modifiers[0] == "Shift";
+            Qt::KeyboardModifier modifier = onlyShift ? Qt::ShiftModifier : Qt::NoModifier;
+
+            // Retrieve scancodes (usually just one, sometimes two,
+            // due to multiple keys sharing the same character)
+            QList<int> scancodes = layoutUtils::findScancodesForCharacter(originalKeyseqLayout, key.at(0), modifier);
+
+            // Warn user if any scancodes where found for this character
+            if (scancodes.isEmpty()) {
+                qWarning() << "Couldn't find any scancodes for character"
+                           << key.at(0) << "on" << keyseqsRawI->byPositionOf
+                           << (onlyShift ? "with" : "without") << "shift."
+                           << "Skipping mapping for" << configKey;
+                continue;
+            }
+
+            // Remove original key sequence
+            keyseqs.clear();
+
+            // Iterate through all found scancodes
+            for (const int& scancode : scancodes) {
+
+                // Get KbdKeyChar
+                const KbdKeyChar *keyChar = layoutUtils::getKbdKeyChar(layout, (unsigned char) scancode, modifier);
+                QChar character(keyChar->character);
+
+                // If key is not dead, reconstruct key sequence with translated character. Otherwise,
+                // warn user about key that won't work.
+                if (!keyChar->isDead) {
+                    QString modifiersString = modifiers.join("+");
```

Comment:
Why not copying the modifiers string list, appending the char as a string, and finally joining the whole list with '+' into a single string? The code would be much cleaner and easier to read:
QStringList youNameIt(modifiers);
youNameIt.append(keyChar);
keyseqs.append(youNameIt.join("+"));